File: review-microshed-testing/src/main/java/de/rieckpil/blog/Person.java
package de.rieckpil.blog;

import javax.json.JsonObject;
import javax.persistence.*;

@Entity
public class Person {

    @Id
    @GeneratedValue
    private Long id;

    @Column(nullable = false)
    private String firstName;

    @Column(nullable = false)
    private String lastName;

    public Person() {
    }

    public Person(JsonObject jsonObject) {
        this.firstName = jsonObject.getString("firstName", "not provided");
        this.lastName = jsonObject.getString("lastName", "not provided");
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }
}

File: spring-boot-with-kotlin/src/main/kotlin/de.rieckpil.blog/Person.kt
package de.rieckpil.blog

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import java.time.LocalDate

@Entity
class Person(
  @Id @GeneratedValue var id: Long? = null,
  @Column(nullable = false) var firstname: String,
  @Column(nullable = false) var lastname: String,
  @Column(unique = true, nullable = false) var username: String,
  @Column(nullable = false) var dayOfBirth: LocalDate,
)
